### PR TITLE
Remove doctrine/common backref and use specific packages instead.

### PR DESCRIPTION
**Pull request by jwage**
Update `doctrine/reflection` from `1.0@dev` to `^1.0` once the v1.0.0 tag is created.

### CODE REVIEW

**Comment by Majkl578** [comment]
Looks good, let's just wait for Reflection 1.0 tag.

Any specific reason why earlier Collections/Annotations/Cache are not allowed?

**Comment by jwage** [comment]
@Majkl578 I can change that. What version is the min I should require? I added them to composer.json with composer require so it just added the latest version.

File: lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php [hunk 1]
```diff
@@ -394,4 +396,22 @@ public function getReflectionService()
...
     }
+
+    /**
+     * Gets the real class name of a class name that could be a proxy.
+     *
+     * @param string $class
+     *
+     * @return string
+     */
+    private function getRealClass($class)
```
Comment (alcaeus) [inline lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php]: Shouldn't this be part of a `doctrine/proxy` implementation?

Comment (Majkl578) [reply lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php]: We don't really want doctrine/proxy, it's to be dropped soon.

Comment (alcaeus) [reply lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php]: Gotcha 👍

Comment (Majkl578) [inline lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php]: We should make this look like:
```php
private static function getRealClass(string $class) : string
```

Comment (jwage) [reply lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php]: @Majkl578 done

**Pull request merged**